A66-otel-stats.md

```diff
@@ -0,0 +1,542 @@
+    serialization.
+*   When new attempts are created on the call along with information on whether
+    the attempt was a transparent retry or not. (Attempts are created after name
+    resolution and after any xDS HTTP filters but before the LB pick.) This is
+    also when it's expected for the `CallAttemptTracer` to be created.
+*   When an attempt ends. This will be needed for future stats around retries
+    and hedging. This information can also be propagated through the
+    `CallAttemptTracer` if the `CallAttemptTracer` keeps a reference to the
+    parent `CallTracer` object.
+*   When the call ends. This along with the call creation call-out allows the
+    `CallTracer` to calculate the call duration.
+
+The following call-outs are needed on the `CallAttemptTracer` -
+
+*   When a new message is sent/received. The message should be in its compressed
+    form.
+*   When the trailing metadata/status is received for the attempt. Receipt of
+    this indicates that the attempt has ended. Implementations may choose to
+    delegate the responsibility of notifying the `CallTracer` about the attempt
+    end to the `CallAttemptTracer`.
+
+The following call-outs are needed on the `ServerCallTracer` -
+
+*   When initial metadata is received by the transport for a call. This
+    indicates the start time of a new call.
+*   When a new message is sent/received. The message should be in its compressed
+    form.
+*   When trailing metadata/status is sent. This call-out should be as close to
+    the transport as possible to be able to capture the total time of the call.
+
+Implementations should allow multiple call/attempt tracers to be registered to a
+single call since there could be multiple plugins registered. For example, there
+could be an OpenCensus and an OpenTelemetry stats plugin registered together. It
+should also allow multiple OpenTelemetry plugins to be registered providing the
+ability to configure the different plugins with different MeterProviders.
+
+A sample implementation of this approach is available in
+[gRPC Core](https://github.com/grpc/grpc/blob/v1.57.x/src/core/lib/channel/call_tracer.h).
+
+### Language-Specific Details
+
+Each language implementation will provide an API for registering an
+OpenTelemetry plugin. Overall, the APIs should have the following capabilities -
+
+*   Allow installing multiple OpenTelemetry plugins.
+*   Allow setting a
+    [MeterProvider](https://opentelemetry.io/docs/specs/otel/metrics/api/#meterprovider)
+    on individual plugins. Implementations should require a MeterProvider being
+    set. A MeterProvider not being set should result in a no-op. Some
+    OpenTelemetry language APIs have a global MeterProvider. gRPC
+    implementations should *NOT* fallback on this global.
+*   Optionally allow enabling/disabling metrics. This would allow optimizations
+    to avoid computation and collection of expensive stats within the gRPC
```

Comment:
In a vacuum this doesn't make sense.  Simply don't register the plugin if you don't want metrics enabled...  But I think the idea is that the same plugin will do tracing in the future, right?

Reply:
Yes

Reply:
I think the idea of this language is if a user wants a subset of the metrics, avoiding computation on the metrics the user doesn't want.

Reply:
So then "Optionally allow enabling/disabling _individual_ metrics" please

Reply:
We removed this since we want it to be barebones for now, we'll consider it in a subsequent gRFC

Reply:
Even better!